geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/kryo/KryoSimpleFeatureFactory.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2015 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.locationtech.geomesa.feature.kryo
+
+import com.google.common.cache.{CacheBuilder, CacheLoader}
+import org.geotools.factory.{CommonFactoryFinder, Hints}
+import org.geotools.feature.AbstractFeatureFactoryImpl
+import org.geotools.feature.simple.SimpleFeatureBuilder
+import org.geotools.filter.identity.FeatureIdImpl
+import org.locationtech.geomesa.utils.text.{ObjectPoolFactory, ObjectPoolUtils}
+import org.opengis.feature.`type`.AttributeDescriptor
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.collection.JavaConversions._
+import scala.ref.SoftReference
+
+class KryoSimpleFeatureFactory extends AbstractFeatureFactoryImpl {
+
+  override def createSimpleFeature(attrs: Array[AnyRef], sft: SimpleFeatureType, id: String) = {
+    val sf = new KryoSimpleFeature(id, sft)
+    sf.setAttributes(attrs)
+    sf
+  }
+
+  override def createSimpleFeautre(attrs: Array[AnyRef], descriptor: AttributeDescriptor, id: String) =
+    createSimpleFeature(attrs, descriptor.asInstanceOf[SimpleFeatureType], id)
+
+}
+
+object KryoSimpleFeatureFactory {
+
+  import scala.collection.mutable.Map
+
+  private val hints = new Hints(Hints.FEATURE_FACTORY, classOf[KryoSimpleFeatureFactory])
+  private val featureFactory = CommonFactoryFinder.getFeatureFactory(hints)
+
+  private val cache = new ThreadLocal[Map[SimpleFeatureType, SoftReference[SimpleFeatureBuilder]]] {
```

Comment:
Yeah, guava caches would work in a thread local. However, I assume there is some overhead since the caches are meant to be shared between threads. This implementation is probably overkill... I did it this way based off of some kryo sample code here: https://github.com/EsotericSoftware/kryo#pooling-kryo-instances, was just trying to make it at fast as possible. I can extract it out into a class, though.